colibri/src/icons.rs

```diff
@@ -119,6 +138,86 @@ async fn query_token_icon_and_extension(
     None
 }
 
+async fn query_uniswap_v3_position_icon(
+    chain_id: u64,
+    token_id: &str,
+    contract_address: &str,
+    globaldb: &globaldb::GlobalDB,
+) -> Option<(Bytes, &'static str)> {
+    let blockchain = SupportedBlockchain::from_chain_id(chain_id)?;
+    let rpc_endpoints = globaldb.get_rpc_nodes(blockchain).await.ok()?;
+    let contract_address = match Address::parse_checksummed(contract_address, None) {
+        Ok(address) => address,
+        Err(_e) => {
+            return None;
+        }
+    };
+    let token_id: U256 = token_id.parse().unwrap();
+
+    // Try all RPC endpoints until one succeeds
+    for rpc_url in rpc_endpoints {
+        let provider = match rpc_url.parse().map(|url| ProviderBuilder::new().on_http(url)) {
+            Ok(provider) => provider,
+            Err(e) => {
+                error!("Failed to setup provider for {}: {}", rpc_url, e);
+                continue;
+            }
+        };
+
+        let contract = IERC721Metadata::new(contract_address, provider);
+        let token_uri = match contract.tokenURI(token_id).call().await {
+            Ok(result) => result._0,
+            Err(e) => {
+                error!("RPC call failed on {}: {}", rpc_url, e);
+                continue;
+            }
+        };
+        let base64_str = match token_uri.strip_prefix("data:application/json;base64,") {
+            Some(data) => data,
+            None => {
+                error!("Invalid token URI format from {}", rpc_url);
+                continue;
+            }
+        };
+        let json_bytes = match BASE64_STANDARD.decode(base64_str) {
+            Ok(bytes) => bytes,
+            Err(e) => {
+                error!("Failed to decode base64 JSON from {}: {}", rpc_url, e);
+                continue;
+            }
+        };
+        let json_data: serde_json::Value = match serde_json::from_slice(&json_bytes) {
+            Ok(json) => json,
+            Err(e) => {
+                error!("Failed to parse JSON from {}: {}", rpc_url, e);
+                continue;
+            }
+        };
+        let image_uri = match json_data.get("image").and_then(|v| v.as_str()) {
+            Some(uri) => uri,
+            None => {
+                error!("No 'image' field in JSON from {}", rpc_url);
+                continue;
+            }
+        };
```

Comment:
is something like this possible?

```rust
if let Ok(json_bytes) = BASE64_STANDARD.decode(base64_str) {
    if let Ok(json_data) = serde_json::from_slice(&json_bytes) {
        if let Some(image_uri) = json_data.get("image").and_then(|v| v.as_str()) {
            // Use image_uri here
        } else {
            error!("No 'image' field in JSON from {}", rpc_url);
        }
    } else {
        error!("Failed to parse JSON from {}: {}", rpc_url, e);
    }
} else {
    error!("Failed to decode base64 JSON from {}: {}", rpc_url, e);
    continue;
}
```

I find it easier to read and uses less lines